arduino/src/arduino.cpp

```diff
@@ -0,0 +1,285 @@
+    std::string fileName = "../dataFromArduino/sensor";
+    std::string fileNameEnd = ".log";
+    fileName.append(std::to_string(place));
+    fileName.append(fileNameEnd);
+    dataLog.open(fileName, std::ios_base::app);
+    if (!dataLog)
+    {
+        return Arduino::ERROR_ARDUINO::ERROR_WITH_FILES;
+    }
+    dataLog << std::time(0) << std::endl;
+    dataLog << data << std::endl;
+    dataLog.close();
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+/**
+ * @brief this is the main function,
+ * will listen to the serial port and interact with it.
+ * 
+ * @return Arduino::ERROR_ARDUINO will return the error code if there is one. 
+ */
+Arduino::ERROR_ARDUINO Arduino::getDataWithWhileLoop()
+{
+    if (openSerial() == ERROR_WITH_SERIALPORT)
+    {
+        return Arduino::ERROR_ARDUINO::ERROR_WITH_SERIALPORT;
+    }
+    while (true)
+    {
+        if (this->serial.available())
+        {
+            uint32_t num = 0;
+            this->serial.readBytes(&num, sizeof(num));
+            switch (num)
+            {
+            case HEADERS::GET_FLOAT:
+                receiveDataFromSensor();
+                continue;
+            case HEADERS::ASK_FOR_OVERALL_STATUS:
+                checkStatusFromArduino();
+                continue;
+            default:
+                continue;
+            }
+            return Arduino::ERROR_ARDUINO::SUCCSESS;
+        }
+    }
+}
+/**
+ * @brief will receive a message and write it to the sensor log
+ * @return Arduino::ERROR_ARDUINO will return the error code if there is one. 
+ */
+Arduino::ERROR_ARDUINO Arduino::receiveDataFromSensor()
+{
+    uint32_t id = 0;
+    this->serial.readBytes(&id, sizeof(id));
+    float f = receiveFloat();
+    this->writeFromBufferToFile(std::to_string(f), id);
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+/**
+ * @brief will receive float from serial port
+ * 
+ * @return float the value that came out.
+ */
+float Arduino::receiveFloat()
+{
+    float value = 0;
+    this->serial.readBytes(&value, sizeof(value));
+    return value;
+}
+/**
+ * @brief will add mock files for the logging
+ */
+void Arduino::addLogFilesForSensors()
+{
+    this->addFile("..//dataFromArduino//sensor0.log");
+    this->addFile("..//dataFromArduino//sensor1.log");
+    this->addFile("..//dataFromArduino//sensor2.log");
+    this->addFile("..//dataFromArduino//sensor3.log");
+    this->addFile("..//dataFromArduino//sensor4.log");
+    this->addFile("..//dataFromArduino//sensor5.log");
+}
+/**
+ * @brief the simplest way to receive a boolean from the arduino
+ * @return bool the value that came from the serial port.
+ */
+bool Arduino::receiveBoolean()
+{
+    bool value = 0;
+    this->serial.readBytes(&value, sizeof(value));
+    return value;
+}
+/**
+ * @brief the function will send the arduino a message to 
+ * start checking all of its modules. It will look for errors
+ * and if there are it will reboot the arduino.
```

Comment:
its not what the function dose

Reply:
the Arduino will check himself and send the result

Reply:
fixed

Reply:
not fixed